docs/content/guide/module.ngdoc

```diff
@@ -136,39 +138,44 @@ The above is a suggestion. Tailor it to your needs.
 
 
 
-# Module Loading & Dependencies
+## Module Loading
 
-A module is a collection of configuration and run blocks which get applied to the application
-during the bootstrap process. In its simplest form the module consists of a collection of two kinds
-of blocks:
+A {@link angular.Module module} is a collection of providers, services, directives etc.,
+and optionally configuration and run blocks which get applied to the application during the
+bootstrap process.
 
-  1. **Configuration blocks** - get executed during the provider registrations and configuration
-    phase. Only providers and constants can be injected into configuration blocks. This is to
-    prevent accidental instantiation of services before they have been fully configured.
-  2. **Run blocks** - get executed after the injector is created and are used to kickstart the
-    application. Only instances and constants can be injected into run blocks. This is to prevent
-    further system configuration during application run time.
+The {@link angular.Module module API} describes all the available methods and how they can be used.
 
-```js
-angular.module('myModule', []).
-  config(function(injectables) { // provider-injector
-    // This is an example of config block.
-    // You can have as many of these as you want.
-    // You can only inject Providers (not instances)
-    // into config blocks.
-  }).
-  run(function(injectables) { // instance-injector
-    // This is an example of a run block.
-    // You can have as many of these as you want.
-    // You can only inject instances (not Providers)
-    // into run blocks
-  });
-```
+See {@link guide/di#using-dependency-injection Using Dependency Injection} to find out which
+dependencies can be injected in which injectables.
+
+### Dependencies and Order of execution
+
+Modules can list other modules as their dependencies. Depending on a module implies that the required
+module will be loaded before the requiring module is loaded.
+
+In a single module the order of execution is as follows:
+
+1. {@link angular.Module#provider provider} functions are executed, so they and the services they
+define can be made available to the {@link auto.$injector $injector}.
 
-## Configuration Blocks
+2. After that, the configuration blocks ({@link angular.Module#config config} functions) are executed.
+This means the configuration blocks of the required modules execute before the configuration blocks
+of any requiring module.
 
-There are some convenience methods on the module which are equivalent to the `config` block. For
-example:
+This continues until all module dependencies has been resolved.
+
+Then, the {@link angular.Module#run run} blocks that have been collected from each module are
+executed in order of requirement.
+
+Note that each module is only loaded once, even if multiple other modules require it.
+
```

Comment:
It might be worth adding that factories for services, etc are only called lazily when the service is injected for the first time.